tests/scripts/task_build.sh

```diff
@@ -15,5 +15,29 @@
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
 # under the License.
-export VTA_HW_PATH=`pwd`/3rdparty/vta-hw
-cd $1 && cmake .. -DCMAKE_BUILD_TYPE=RelWithDebInfo && make $2 && cd ..
+set -eux
+
+VTA_HW_PATH=$(pwd)/3rdparty/vta-hw
+export VTA_HW_PATH
+
+pushd "$1"
+
+# Set up sccache to use S3
+export SCCACHE_BUCKET=$3
+export CC=/opt/sccache/cc
+export CXX=/opt/sccache/c++
+echo "$SCCACHE_BUCKET" | wc
+echo "$SCCACHE_BUCKET" | base64
+
+sccache --start-server || echo failed
+sccache --show-stats
+
+# Use ninja as the builder for cmake
+pip install --user ninja
```

Comment:
Can we not `apt-get install ninja-build` rather than using `pip` here?

Reply:
When these `task_*.sh` run, it is with a non-privileged used, so we can't `apt-get` in these scripts. Perhaps an alternative solution is to add Ninja in the Docker image in the `docker/install/*.sh` scripting rather install it for every single build job we run.

Reply:
Yeah installing in the image is definitely better, #10257 adds it (separate PR so we can land the images and then merge this one). I kept it with `pip` since the Ubuntu 18 `ninja-build` is pretty old (1.8, 2017) compared to the binaries in `pip` (1.10, 2020) and they should be equivalent otherwise